File: app/src/main/java/com/example/ui/main/MainPresenter.java
package com.example.ui.main;

import com.example.data.DataManager;
import com.example.data.model.User;
import com.example.data.response.ReminderResponse;
import com.example.injection.ConfigPersistent;
import com.example.ui.base.BasePresenter;
import com.google.gson.Gson;
import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;
import javax.inject.Inject;

@ConfigPersistent
public class MainPresenter extends BasePresenter<MainMvpView> {

    private final DataManager mDataManager;

    @Inject
    public MainPresenter(DataManager dataManager) {
        mDataManager = dataManager;
    }

    public void loadMain() {
        getMvpView().showLoading();
        mDataManager.getDataList("ongoing")
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe(new Observer<ReminderResponse>() {
                    @Override
                    public void onSubscribe(final Disposable d) {
                        getDisposableContainer().add(d);
                    }

                    @Override
                    public void onNext(final ReminderResponse response) {
                        if (response != null && response.getStatus() != null) {
                            if (response.getStatus().equals("success")) {
                                getMvpView().onSuccess(response.reminderList);
                                User user = new Gson().fromJson(mDataManager.getSavedUser(), User.class);
                                getMvpView().onSetUser(user);
                            } else {
                                getMvpView().showError(response.getMessage());
                            }
                        }
                        getMvpView().hideLoading();
                    }

                    @Override
                    public void onError(final Throwable e) {
                        getMvpView().showError(e.getMessage());
                        getMvpView().hideLoading();
                    }

                    @Override
                    public void onComplete() {}
                });
    }

}

File: app/src/main/java/com/example/ui/main/MainPresenter.kt
package com.example.ui.main

import com.example.base.BasePresenter
import com.example.data.DataManager
import com.example.data.model.User
import com.example.injection.ConfigPersistent
import com.google.gson.Gson
import javax.inject.Inject

class MainPresenter @Inject constructor(private val mDataManager: DataManager) : BasePresenter<MainMvpView?>() {

    fun loadUser() {
        Gson().fromJson(mDataManager.savedUser, User::class.java)?.run {
            mvpView?.onSetUser(this)
        }
    }

    /*fun loadMain() {
        mvpView?.showLoading()
        mDataManager.getDataList("ongoing")
            .observeOn(AndroidSchedulers.mainThread())
            .subscribeOn(Schedulers.io())
            .subscribe(object : Observer<ReminderResponse> {
                override fun onSubscribe(d: Disposable) {
                    disposableContainer.add(d)
                }

                override fun onNext(response: ReminderResponse) {
                    if (response.status != null) {
                        if (response.status == "success") {
                            mvpView?.onSuccess(response.reminderList)
                            val user = Gson().fromJson(mDataManager.savedUser, User::class.java)
                            mvpView?.onSetUser(user)
                        } else {
                            mvpView?.showError(response.message)
                        }
                    }
                    mvpView?.hideLoading()
                }

                override fun onError(e: Throwable) {
                    mvpView?.showError(e.message)
                    mvpView?.hideLoading()
                }

                override fun onComplete() {}
            })
    }*/
}